applications/tari_console_wallet/src/main.rs

```
@@ -234,24 +233,24 @@ async fn setup_wallet(
         }
     })?;
 
-    debug!(target: LOG_TARGET, "Setting peer seeds");
-
-    // TODO update this to come from an explicit config field. This will be replaced by gRPC interface.
-    if !config.peer_seeds.is_empty() {
-        let seed_peers = parse_peer_seeds(&config.peer_seeds);
-        let random_seed_peer = OsRng.next_u32() as usize % seed_peers.len();
-        wallet
-            .set_base_node_peer(
-                seed_peers[random_seed_peer].public_key.clone(),
-                seed_peers[random_seed_peer]
-                    .addresses
-                    .first()
-                    .expect("The seed peers should have an address")
-                    .to_string(),
-            )
-            .await
-            .map_err(|e| ExitCodes::WalletError(format!("Error setting wallet base node peer. {}", e)))?;
-    }
+    debug!(target: LOG_TARGET, "Setting base node peer");
```

Are we sure that if no base node is set that the wallet will error out on startup? A wallet can run and interact with other wallets without speaking to base nodes. It is not ideal but I am not sure its a full on error show stopper? Thoughts?

---

Good point, probably best to separate setting peer seeds from the base node peer? 

In that case, if no base node is set then default to one of the seeds?

---

I think this is a point to discuss a bit further. I like this proposed strategy, I think it is the most plug and play path to the wallet just working out the box.

However based on our discussion yesterday we might not want to default to a seed peer if we feel that users will want to be careful about which base node they connect to we might only want to connect to explicitly chosen base nodes? I think perhaps that these will be a small percentage of power users,

---

For now, I'm going to implement the fallback to a random seed peer for ease of use, with logging info and documentation. Privacy-oriented users will of course be able to override the setting in config. We can then iterate on this later if necessary. 

Thanks for the feedback @philipr-za !